### PR TITLE
Edits to add a merge CLI and associated acceptance tests

### PR DESCRIPTION
**Pull request by gavinevans**
Description
This PR:
* Adds a merge CLI that merges multiple files into a single file.
* Adds associated acceptance tests.

Testing:
 - [x] Ran tests and they passed OK
 - [x] Added new tests for the new feature(s)

### CODE REVIEW

**Comment by benfitzpatrick** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- improver_tests/acceptance/test_merge.py  1
- improver/cli/merge.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=5073172&bid=16690468)

File: improver/cli/merge.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    Args:
+        cubes (list of iris.cube.Cube):
+            Cubes to be merged.
+
+    Returns:
+        iris.cube.Cube:
+            A merged cube.
+    """
+    from iris.cube import CubeList
+    return CubeList(cubes).merge_cube()
```
Comment (bayliffe) [inline improver/cli/merge.py]: I think we should use our own MergeCubes plugin so that it can handle mismatched attributes (like history) etc. Thoughts?

**Comment by benfitzpatrick** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- improver_tests/acceptance/test_merge.py  1
- improver/cli/merge.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=5073172&bid=16690468)

File: improver/cli/merge.py (hunk 1 above)
Comment (gavinevans) [reply improver/cli/merge.py]: Good idea. I've updated this PR to use the `MergeCubes` plugin.

**Pull request merged**